viewing_party/party.py

```diff
@@ -1,23 +1,156 @@
 # ------------- WAVE 1 --------------------
 
+from tests.test_constants import MOVIE_TITLE_1
+
+
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if not title:
+        return None
+    elif not genre:
+        return None
+    elif not rating:
+        return None
+    else:
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+        return movie_dict
+
+def add_to_watched(user_data, movie):
+    movie_list = user_data["watched"]
+    movie_list.append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    watch_list= user_data["watchlist"]
+    watch_list.append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    user_data_list = user_data["watchlist"]
+    user_data_watched = user_data["watched"]
+    for movie_details in user_data["watchlist"]:
+        if movie_details["title"] == title:
+            user_data_list.remove(movie_details)
+            user_data_watched.append(movie_details)
+            return user_data
+        else:
+            continue
+    return user_data

```

Comment:
In the future, try to avoid modifying a list as you're looping over it. You still pass the tests here, but it can lead to some strange bugs! It would be best to modify the list after the loop is finished.